### PR TITLE
[7465] Rename RequiresTypeResolution to RequiresFullAnalysis

### PR DESCRIPTION
**Pull request by alXmas**
Fixes: https://github.com/detekt/detekt/issues/7465

### CODE REVIEW

**Comment by 3flex** [comment]
Both will require doc updates too.

**Comment by BraisGabin** [comment]
I'm still not fully on board with my pr so if @ this one updates the docs I'm OK with merging this before.

**Comment by alXmas** [comment]
> Warnings
> ⚠️ 	
> 
> It looks like you're editing the **un-versioned copy** of our website. This affects only users on the 'next' version of detekt, and it's correct only if you intend to document a **future change or feature**. If you intended to make a change also for the **current** version of detekt, please make sure you edit the equivalent file inside `website/versioned_docs/` as well.
> 	Messages
> 📖 	
> 
> Thank you very much for making [our website](https://detekt.dev/) better ❤️!
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [b331865](https://github.com/detekt/detekt/commit/b331865c1227108e3506504403866acc781a629f)

I run `yarn generate-and-build` in website folder, but nothing added/updated :(

File: website/src/pages/changelog.md [hunk 1]
```diff
@@ -397,7 +397,7 @@ See all issues at: [1.23.0](https://github.com/detekt/detekt/milestone/88)
 -   We added a new ruleset called `detekt-rules-libraries` containing rules mostly useful for Library Authors - We moved the following rules inside `ForbiddenPublicDataClass`, `LibraryCodeMustSpecifyReturnType`, `LibraryEntitiesShouldNotBePublic` this new ruleset - See Migration below on how to migrate [#5360](https://github.com/detekt/detekt/pull/5360)
 -   We added support for JVM toolchain. This means that detekt will now respect the JDK toolchain you specify on your Gradle configuration. You will also be able to specify a custom JDK home with the `--jdk-home` CLI parameter - [#5269](https://github.com/detekt/detekt/pull/5269)
 -   Improvement for Type Resolution
-    -   We will now skip rules annotated with `@RequiresTypeResolution` when without Type Resolution - [#5176](https://github.com/detekt/detekt/pull/5176)
```
Comment (cortinico) [inline website/src/pages/changelog.md:400]: Please undo the changes to the CHANGELOG file

**Review by cortinico — CHANGES_REQUESTED** [review]

**Comment by cortinico** [comment]
> I run `yarn generate-and-build` in website folder, but nothing added/updated :(

You can disregard the warning this time as you actually need to update **only** the unversioned docs (they will appear once 2.x is out)

File: website/src/pages/changelog.md (hunk 1 above)
Comment (alXmas) [reply website/src/pages/changelog.md]: Fixed! :)

**Review by cortinico — APPROVED** [review]

**Review by BraisGabin — APPROVED** [review]

**Pull request merged**